pkg/config/replica_config.go

```diff
@@ -240,17 +240,19 @@ func (c *ReplicaConfig) FixMemoryQuota() {
 func GetSinkURIAndAdjustConfigWithSinkURI(
 	sinkURIStr string,
 	config *ReplicaConfig,
-) (*url.URL, error) {
+) (*url.URL, *ReplicaConfig, error) {
```

Comment:
Maybe it is better to make `replicaConfig` to provide some thread-safe API, and separate readonly and writable API. 
`GetSinkURIAndAdjustConfigWithSinkURI` seems mix both read and write in one function.

Reply:
I changed how to fix this problem. It is better  to not change the value of replicaConfig while changefeed is running. Instead, it should be adjusted in the create or update API.